firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java

```diff
@@ -205,9 +195,8 @@ public CollectionReference collection(@NonNull String collectionPath) {
    * @return A Task that will be resolved when the write finishes.
    */
   @NonNull
-  @PublicApi
   public Task<Void> update(
-      @NonNull String field, @Nullable Object value, Object... moreFieldsAndValues) {
+      @NonNull String field, @Nullable Object value, @NonNull Object... moreFieldsAndValues) {
```

Comment:
This seems incorrect. `null` is a valid value in here. Does the `@NonNull` apply to elements of the argument list or to the argument list itself?

Reply:
It is for the argument list itself, there is no way to annotate members of the list unfortunately, so the kotlin signature will be
```kotlin
fun update(field: String, value: Any?, vararg moreFieldsAndValues: Any!) // note the !
```

Furthermore I only added this to silence the lint check, as this annotation has no effect on the above signature.

I removed these annotations now and updated the check to ignore varargs parameters.